### PR TITLE
feat: Added authentication logic with miner fix

### PR DESCRIPTION
**Pull request by VishalMinj**
## Added:
- Login and Signup pages and links
- Also added logout Link
- Toast messages to certain validation
- Minor JS for toast and password visibility
## Updated:
- Assigned login required to the required view
- Adjusted the redirect urls in settings
- Adjusted the ux of navbar slightly 

Note: I've also registered the ticket model which was previously not registered in admin site

### CODE REVIEW

**Comment by VishalMinj** [comment]
One thing I'd also like to mention is the css I've added is not responsive exactly

**Comment by eleensmathew** [comment]
Can you resolve the conflicts?

File: user/templates/base.html [hunk 1]
```diff
@@ -6,19 +6,30 @@
...
-        {% endif %}
-        <a class="nav-item nav-link" href="{% url 'user:movie_list' %}">Movie List</a>
-    </nav>
+    <header>
+        <div class="logo">Movie Screen</div>
+        <nav>
+            {% if request.user.is_authenticated %}
+            {% if request.user.is_staff %}
+            <a class="nav-item nav-link " href="{% url 'user:add_movie' %}">Add Movie</a>
+            {% endif %}
```
Comment (eleensmathew) [inline user/templates/base.html:20]: Can you add a nav-item for add-show as well ? I'll give you bonus

Comment (VishalMinj) [reply user/templates/base.html:20]: okay let me see, but that will make the navbar look a bit cluttered

**Comment by eleensmathew** [comment]
Its still unable to merge

**Comment by VishalMinj** [comment]
you merged a pr right? Might need to pull and update first

**Comment by eleensmathew** [comment]
Yup you need to

**Comment by VishalMinj** [comment]
Merged and updated it, Except for this one password toggle hover everything seems to be fine

image

**Pull request merged**